code/Modules/Items/Carriables/ItemTorchlight.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Sandbox;
+using ssl.Modules.Items.Data;
+
+namespace ssl.Modules.Items.Carriables
+{
+    public partial class ItemTorchlight : Item
+    {
+        private const int Range = 512;
+        private const float Falloff = 1.0f;
+        private const float LinearAttenuation = 0.0f;
+        private const float QuadraticAttenuation = 1.0f;
+        private const int Brightness = 2;
+        private SpotLightEntity worldLight;
+        private SpotLightEntity viewLight;
+        private const int InnerConeAngle = 20;
+        private const int OuterConeAngle = 40;
+        private const float FogStength = 1.0f;
+        public ItemTorchlight()
+        {
+        }
+        
+        public ItemTorchlight(ItemData data) : base(data)
+        {
+        }
+        
+        //TODO Replace with the NeckCamera 
+        public override string ViewModelPath => "weapons/rust_flashlight/v_rust_flashlight.vmdl";
+        
+        private static Vector3 LightOffset => Vector3.Forward * 10;
+        [Net, Local, Predicted]
+        private bool LightEnabled { get; set; } = true;
```

Comment:
Why Local ? Other players need to see if the torchlight is on 
We'll need to test this